src/lib/sticky-header/sticky-header.spec.ts

```diff
@@ -0,0 +1,175 @@
+  });
+
+  it('should be able to find stickyParent', () => {
+    expect(stickyHeaderDir.stickyParent).not.toBe(null);
+  });
+
+  it('should be able to find scrollableContainer', () => {
+    expect(stickyHeaderDir.upperScrollableContainer).not.toBe(null);
+  });
+
+  it('should stick in the right place when scrolled to the top of the container', fakeAsync(() => {
+    let scrollableContainerTop = stickyHeaderDir.upperScrollableContainer
+      .getBoundingClientRect().top;
+    expect(stickyHeaderDir.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+    tick(0);
+
+    // Scroll the scrollableContainer up to stick
+    fixture.componentInstance.scrollDown();
+    tick(100);
+
+    expect(stickyHeaderDir.element.getBoundingClientRect().top).toBe(scrollableContainerTop);
+  }));
+
+  it('should unstuck when scrolled off the top of the container', fakeAsync(() => {
+    let scrollableContainerTop = stickyHeaderDir.upperScrollableContainer
+      .getBoundingClientRect().top;
+    expect(stickyHeaderDir.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+    tick(0);
+
+    // Scroll the scrollableContainer up to stick
+    fixture.componentInstance.scrollDown();
+    tick(100);
+
+    expect(stickyHeaderDir.element.getBoundingClientRect().top).toBe(scrollableContainerTop);
+
+    // Scroll the scrollableContainer down to unstuck
+    fixture.componentInstance.scrollBack();
+    tick(100);
+
+    expect(stickyHeaderDir.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+
+  }));
+});
+
+describe('sticky-header with positioning supported', () => {
+  let fixture: ComponentFixture<StickyHeaderTest>;
+  let testComponent: StickyHeaderTest;
+  let stickyElement: DebugElement;
+  let stickyParentElement: DebugElement;
+  let stickyHeaderDir: CdkStickyHeader;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ OverlayModule, PlatformModule, StickyHeaderModule ],
+      declarations: [StickyHeaderTest],
+      providers: [
+        {provide: STICKY_HEADER_SUPPORT_STRATEGY, useValue: true},
+      ],
+    });
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(StickyHeaderTest);
+    fixture.detectChanges();
+    testComponent = fixture.debugElement.componentInstance;
+    stickyElement = fixture.debugElement.query(By.directive(CdkStickyHeader));
+    stickyParentElement = fixture.debugElement.query(By.directive(CdkStickyRegion));
+    stickyHeaderDir = stickyElement.injector.get<CdkStickyHeader>(CdkStickyHeader);
+  });
+
+  it('should find sticky positioning is applied', () => {
+    let position = window.getComputedStyle(stickyHeaderDir.element).position;
+    expect(position).not.toBe(null);
+    if (position != null) {
+      expect(/sticky/i.test(position)).toBe(true);
+    }
+  });
+});
+
+@Component({
+  template: `
+    <div cdk-scrollable style="text-align: center;
```

Comment:
:+1:  Added comments on why we need to apply these styles. And put set styles via the `styles` configuration for the component.